src/docfx/docset/Document.cs

```diff
@@ -77,6 +77,28 @@ public string ReadText()
             }
         }
 
+        public override int GetHashCode()
+        {
+            // todo: add docset for calculation
+            return StringComparer.OrdinalIgnoreCase.GetHashCode(FilePath);
+        }
+
+        public bool Equals(Document other)
+        {
+            if (other == null)
+            {
+                return false;
+            }
+
+            // todo: add docset for comparing
+            return string.Equals(other.FilePath, FilePath, StringComparison.OrdinalIgnoreCase);
+        }
+
+        public override bool Equals(object obj)
+        {
+            return Equals((Document)obj);
```

Comment:
if the type of obj is not `Document` please return `false` not throw exception

Reply:
OK

---
In reply to: [184587123](https://github.com/dotnet/docfx/pull/2696#discussion_r184587123) [](ancestors = 184587123)